### DISCUSSION TITLE
Where is the implementation of `CancellationPromise` defined?

### DISCUSSION THREAD

**Discussion by Cyberboss** [dotnet/runtime#88228] [Q&A] [answered]
I was curious about the implementation details of `Task.WaitAsync` but came to a dead end with this struct. 

https://github.com/dotnet/runtime/blob/cdd7566751185d8330b1229b5b7aeb64605fbbf2/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Future.cs#L616

Is it an intrinsic?

**Comment by Joe4evr** [comment, marked as answer]
[Looks like it's defined in `Task.cs`.](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs,24fede97eaa4d8d0)

Also PROTIP: Using the [.NET Source Browser](https://source.dot.net/) (as I did) makes looking for things like this a lot easier.

**Reply by Cyberboss** [reply]
Oh, there it is. What's weird is it absolutely _refuses_ to show up in a GitHub search:
![image](https://github.com/dotnet/runtime/assets/8171642/8214721d-9003-4f68-a279-2763e4726bbe)

**Reply by Joe4evr** [reply]
Yeah, but Github search is the absolute worst.
